docs/developer_guide/debugging.rst

```diff
@@ -0,0 +1,77 @@
+Debugging
+=========
+
+This pages contains some Mantid Imaging specific debugging tips.
+
+Getting Debug Output
+--------------------
+
+There are several options that can be used so that Mantid Imaging outputs more information as it runs, for example::
+
+  python -X faulthandler -W default -m mantidimaging --log-level DEBUG
+
+These options:
+
+* Enable the :py:mod:`faulthandler` to dump a traceback in the event of python crash.
+* Cause python :py:mod:`warnings` to be printed.
+* Set the Mantid Imaging :py:mod:`log level <logging>` to DEBUG.
+
+Logs are written using the Python :py:mod:`logging` module. Mantid Imaging files should access the logger using::
+
+  from logging import getLogger
+  ...
+  LOG = getLogger(__name__)
+
+and then output using::
+
+  LOG.debug(f'Loaded metadata from: {metadata_filename}')
+
+To avoid large amounts of output, logging messages should only be merged if they are likely to be useful for general debugging.
+
+Debugging Github Actions
+------------------------
+
+When errors happen on Github actions it is best to try to reproduce them locally. When that is not possible it may be required to debug them by creating runs by adding commits to a test pull request. In messy cases, it can be best to create a clone of the mantid imaging repository into your Github user account. If you are doing this on the main repository, put :code:`DONTMERGE` in the PR title and commit messages to prevent any accidental merging.
+
+It is usually useful to disable any actions that are not relevant to your issue. On your branch you can delete any un-needed workflows::
+
+  git rm .github/workflows/windows.yml ...
+  git commit -m "DONTMERGE: disable workflows"
+
+If the failure is in a specific test, pytest can be told to only run those by adding a :code:`-k` option::
+
+  - name: pytest
+  timeout-minutes: 10
+  shell: bash -l {0}
+  run: |
+    xvfb-run --auto-servernum python -m pytest --cov --cov-report=xml -n auto -o log_cli=true --ignore=mantidimaging/eyes_tests --durations=10 -k test_parse_metadata_file
+
+
+
+
+In the case of unreliable test failures, it can be useful to repeat part of the workflow. The :code:`if: always()` means that each step will run even if previous ones have failed::
+
+      - name: GUI Tests System 1
+        if: always()
```

Comment:
This is a useful section, thank you for putting this in here. I do think that if we are going to mention `if: always()`, we should also mention the use of `if: success() || failure()` 

`always()` "_Causes the step to always execute, and returns true, even when canceled. A job or step will not run when a critical failure prevents the task from running. For example, if getting sources failed._"
`success() || failure()` will stop running when manually cancelled.

Reply:
https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions is a useful resource we could point devs in

Reply:
`if: '!cancelled()'` would also achieve the same as `if: success() || failure()`, quotes are needed so it's not interpreted as YAML

Reply:
Thanks. I've changed it to  `if: success() || failure()`